crates/nomination/src/ext.rs

```diff
@@ -0,0 +1,95 @@
+#[cfg(test)]
+use mocktopus::macros::mockable;
+
+#[cfg_attr(test, mockable)]
+pub(crate) mod security {
+    use frame_support::dispatch::DispatchResult;
+    use primitive_types::H256;
+
+    pub fn get_secure_id<T: security::Config>(id: &T::AccountId) -> H256 {
+        <security::Module<T>>::get_secure_id(id)
+    }
+
+    pub fn ensure_parachain_status_running<T: security::Config>() -> DispatchResult {
+        <security::Module<T>>::ensure_parachain_status_running()
+    }
+}
+
+#[cfg_attr(test, mockable)]
+pub(crate) mod vault_registry {
+    use crate::DOT;
+    pub use ::vault_registry::VaultStatus;
+    pub use frame_support::dispatch::{DispatchError, DispatchResult};
+    use sp_std::vec::Vec;
+    use vault_registry::LiquidationTarget;
+
+    pub fn get_backing_collateral<T: vault_registry::Config>(vault_id: &T::AccountId) -> Result<DOT<T>, DispatchError> {
+        <vault_registry::Module<T>>::get_backing_collateral(vault_id)
+    }
+
+    pub fn liquidate_vault_with_status<T: vault_registry::Config>(
+        vault_id: &T::AccountId,
+        status: VaultStatus,
+    ) -> Result<DOT<T>, DispatchError> {
+        <vault_registry::Module<T>>::liquidate_vault_with_status(vault_id, status)
+    }
+
+    pub fn increase_backing_collateral<T: vault_registry::Config>(
+        vault_id: &T::AccountId,
+        amount: DOT<T>,
+    ) -> DispatchResult {
+        <vault_registry::Module<T>>::increase_backing_collateral(vault_id, amount)
+    }
+
+    pub fn decrease_backing_collateral<T: vault_registry::Config>(
+        vault_id: &T::AccountId,
+        amount: DOT<T>,
+    ) -> DispatchResult {
+        <vault_registry::Module<T>>::decrease_backing_collateral(vault_id, amount)
+    }
+
+    pub fn lock_additional_collateral_from_address<T: vault_registry::Config>(
+        vault_id: &T::AccountId,
+        collateral: DOT<T>,
+        depositor_id: &T::AccountId,
+    ) -> DispatchResult {
+        <vault_registry::Module<T>>::try_lock_additional_collateral_from_address(vault_id, collateral, depositor_id)
```

Comment:
It would only be necessary if nominated collateral were stored in a separate field in the vault registry. However, this would require changes everywhere `backing_collateral` is used (because you'd need to add the `nominated` collateral to that value).

The current implementation tracks nominated collateral in the `nomination` crate and deposits to the same parachain "account" that the vault deposits to. This way, usage of `backing_collateral` remains unaltered. Direct withdrawals through the `vault_registry`'s dispatchable function are prevented when nomination is enabled, to prevent malicious withdrawals